### ISSUE TITLE
FlatFileDriver::popMessage should throw an exception if not enough permissions to rename the files

### ISSUE THREAD

**Issue by karser** [bernardphp/bernard#382]
I know it sounds obvious but I just have spent an hour figuring out why it doesn't consume the messages that I produce using file driver. 

ls -alh app/cache/dev/bernard/my-queue-name
```
drwxr-xr-x+ 2 www-data www-data 4,0K May 28 12:18 .
drwxr-xr-x+ 3 www-data www-data 4,0K Apr 17 12:56 ..
-rwxr-----+ 1 www-data www-data 2,4K Apr 17 12:56 1.job
-rwxr-----+ 1 www-data www-data 3,2K May 28 11:25 2.job
-rwxr-----+ 1 www-data www-data 3,2K May 28 11:31 3.job
-rwxr-----+ 1 www-data www-data 3,2K May 28 11:36 4.job
-rwxr-----+ 1 www-data www-data 1,4K May 28 11:37 5.job
-rwxr-----+ 1 www-data www-data 7,9K May 28 12:18 6.job
```

The answer was simple permissions error (www-data vs cli user) but shouldn't it throw InsufficientPermissionsException rather than silently failing?
![image](https://user-images.githubusercontent.com/1675033/40608541-74270e90-6274-11e8-925b-544f741f31af.png)

**Comment by henrikbjorn** [comment]
I guess this check should be there in the constructor or when the queue is creted for the first time.

**Comment by karser** [comment]
How about the case where the queue directory was empty - so validation in constructor passed. Then new messages files are created with `www-data` owner. And the CLI user will encounter this at the processing time in `popMessage`, so `@rename` will continue silently failing.

I was suggesting something like that:
![image](https://user-images.githubusercontent.com/1675033/40611577-8deb090c-627f-11e8-9ce3-4151f7169667.png)

**Comment by henrikbjorn** [comment]
Maybe it should but since the file driver is only for development purposes it does not seem good use of power to constantly check for permissions.

**Comment by karser** [comment]
`@rename` is called either way so no additional checks are needed. The problem is that it's silently failing by its nature.

**Comment by henrikbjorn** [comment]
Submit a PR

**Comment by sagikazarmark** [comment]
Fixed in #383
